### PR TITLE
[aws-cognito-identity-js] add AWS Cognito Identity lib

### PR DESCRIPTION
**Pull request by upgradingdave**
New package:

**Extern:** Generated

Depends on jsbn and sjcl libraries.

This provides 2 namespaces via custom deps.clj: 
- cljsjs.aws-cognito-sdk
- cljsjs.aws-cognito-identity-js

### CODE REVIEW

File: aws-cognito-identity-js/build.boot [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+(def +lib-version+ "1.11.0")
+(def +version+ (str +lib-version+ "-1"))
+
+(task-options!
+  push {:ensure-clean false}
+  pom  {:project     'cljsjs/aws-cognito-identity-js
+        :version     +version+
+        :description "Amazon Cognito Identity SDK for JavaScript"
+        :url         "https://github.com/aws/amazon-cognito-identity-js/blob/master/dist/amazon-cognito-identity.min.js"
```
Comment (martinklepsch) [inline aws-cognito-identity-js/build.boot]: I think a URL to the main site about cognito would be more useful here. :)

File: aws-cognito-identity-js/build.boot [hunk 2]
```diff
@@ -0,0 +1,47 @@
+(set-env!
+  :resource-paths #{"resources"}
+  :dependencies '[[cljsjs/boot-cljsjs "0.5.2" :scope "test"]])
```
Comment (miikka) [inline aws-cognito-identity-js/build.boot]: If you added `cljsjs/jsbn` and `cljsjs/sjcl` as dependencies here, then the users of this package wouldn't need to explicitly depend on them.

**Comment by rarous** [comment]
There is missing whole part of extern file for the `amazon-cognito-identity-js`:

```
AWSCognito.CognitoIdentityServiceProvider.AuthenticationDetails = function () {};
AWSCognito.CognitoIdentityServiceProvider.AuthenticationDetails.prototype = {
  "getUsername": function () {},
  "getPassword": function () {},
  "getValidationData": function () {}
};

AWSCognito.CognitoIdentityServiceProvider.AuthenticationHelper = function () {};
AWSCognito.CognitoIdentityServiceProvider.AuthenticationHelper.prototype = {
  "getSmallAValue": function () {},
  "getLargeAValue": function () {},
  "generateRandomSmallA": function () {},
  "generateRandomString": function () {},
  "getRandomPassword": function () {},
  "getSaltDevices": function () {},
  "getVerifierDevices": function () {},
  "generateHashDevice": function () {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoAccessToken = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoAccessToken.prototype = {
  "getJwtToken": function() {},
  "getExpiration": function() {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoIdToken = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoIdToken.prototype = {
  "getJwtToken": function() {},
  "getExpiration": function() {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoRefreshToken = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoRefreshToken.prototype = {
  "getToken": function() {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoUser = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoUser.prototype = {
  "getSignInUserSession": function() {},
  "getUsername": function() {},
  "getAuthenticationFlowType": function() {},
  "setAuthenticationFlowType": function() {},
  "authenticateUser": function() {},
  "completeNewPasswordChallenge": function() {},
  "getDeviceResponse": function() {},
  "confirmRegistration": function() {},
  "sendCustomChallengeAnswer": function() {},
  "sendMFACode": function() {},
  "changePassword": function() {},
  "enableMFA": function() {},
  "disableMFA": function() {},
  "deleteUser": function() {},
  "updateAttributes": function() {},
  "getUserAttributes": function() {},
  "getMFAOptions": function() {},
  "deleteAttributes": function() {},
  "resendConfirmationCode": function() {},
  "getSession": function() {},
  "refreshSession": function() {},
  "cacheTokens": function() {},
  "cacheDeviceKeyAndPassword": function() {},
  "getCachedDeviceKeyAndPassword": function() {},
  "clearCachedDeviceKeyAndPassword": function() {},
  "clearCachedTokens": function() {},
  "getCognitoUserSession": function() {},
  "forgotPassword": function() {},
  "confirmPassword": function() {},
  "getAttributeVerificationCode": function() {},
  "verifyAttribute": function() {},
  "getDevice": function() {},
  "forgetSpecificDevice": function() {},
  "forgetDevice": function() {},
  "setDeviceStatusRemembered": function() {},
  "setDeviceStatusNotRemembered": function() {},
  "listDevices": function() {},
  "globalSignOut": function() {},
  "signOut": function() {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoUserAttribute = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoUserAttribute.prototype = {
  "getValue": function () {},
  "setValue": function () {},
  "getName": function () {},
  "setName": function () {},
  "toString": function () {},
  "toJSON": function () {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoUserPool = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoUserPool.prototype = {
  "getUserPoolId": function () {},
  "getClientId": function () {},
  "getParanoia": function () {},
  "setParanoia": function () {},
  "signUp": function () {},
  "getCurrentUser": function () {}
};

AWSCognito.CognitoIdentityServiceProvider.CognitoUserSession = function () {};
AWSCognito.CognitoIdentityServiceProvider.CognitoUserSession.prototype = {
  "getIdToken": function () {},
  "getRefreshToken": function () {},
  "getAccessToken": function () {},
  "isValid": function () {}
};

AWSCognito.CognitoIdentityServiceProvider.DateHelper = function () {};
AWSCognito.CognitoIdentityServiceProvider.DateHelper.prototype = {
  "getNowString": function () {}
};
```

**Comment by upgradingdave** [comment]
Thanks @rarous, just curious how did you figure out that was missing from externs? Do you use a extern generator? Or did you notice just by looking thru the source code?

**Comment by rarous** [comment]
@upgradingdave According to API docs there much more missing. The generated extern will need much more love.

**Comment by Deraen** [comment]
Closing due to inactivity. Please reopen if you want to finish this PR.

**Pull request closed without merging**